feat(admin): configure facility

bodnara opened the pull request:
* added option for create & configure when creating facility
* created configuration guide that allows to set:
  * hosts
  * owners
  * preconfigure services
  * required attributes(optional)
  * destinations(optional)
  * managers

xkostka2 reviewed (changes requested):
![image](https://user-images.githubusercontent.com/47856862/143879652-5e6a3680-e092-4685-aade-30871f0f44f8.png)
Got this error when getting attributes for service at the "Configure services" tab

Please disable routing in the "Services" at "Summary" tab.

The warning which shows that service destination does not belong to any assigned service is almost never displayed (I managed to trigger it only once by accident)

Would it be possible to at least display some warning (or disable routing to URL with configuration) when the user doesn't come there from facility creation?

xkureck reviewed (changes requested):
![image](https://user-images.githubusercontent.com/32871167/144117602-34596dfa-65ab-4d9b-83a9-1d3274a5861f.png)
hide the checkboxes in the table they are useless

![image](https://user-images.githubusercontent.com/32871167/144118446-71a993cb-f915-4cda-adc9-8b73a60caf54.png)
So on this screenshot
1. when I change some attributes but don't save them when I click on next step no dialog appears
2. its kind of weird to show here attributes that you cant change(eg. Max UID) can we do something about it? maybe its a task for backend but I guess in first step we can also hide it on frontend
![image](https://user-images.githubusercontent.com/32871167/144118953-890e6a21-8a77-4874-9794-e4de4918776e.png)
1. apply right color for the checkboxes (in general)
2. specifically to this checkbox - I think that you should reset to default what is checked in this checkbox because its confusing when I check something and then add some service and the when I get back to this page the same set of services are chosen, 


to discuss on the review or daily scrum - picking of the services plus picking services packages is in my opinion weird and maybe should be redesigned+

also consistency - on summary of managers you have title and on the others you don't, I must say that I quite like that the tabs have titles when you open them
![image](https://user-images.githubusercontent.com/32871167/144130762-62e20b35-a72e-4487-85a4-348b18ee92a1.png)
![image](https://user-images.githubusercontent.com/32871167/144130829-df9c6861-a2ea-404a-9b10-4801799cf13e.png)

and fix some eslint warnings

xkureck commented on apps/admin-gui/src/app/facilities/pages/facility-configuration-page/facility-configuration-page.component.html:
```diff
@@ -0,0 +1,232 @@
+<div class="facility-theme">
+  <h1 class="page-title">
```
this suggestion

xkureck reviewed (changes requested):
![image](https://user-images.githubusercontent.com/32871167/144390207-bb0cc444-936c-40cd-8132-76f621e5ddca.png)
nothing happens when i click on skip

![image](https://user-images.githubusercontent.com/32871167/144392007-c28149e7-3ef2-42ea-8b99-7ebad5d7ff99.png)
missing subtitle of the page - "Services", also please hide the column "script"

![image](https://user-images.githubusercontent.com/32871167/144392262-6a3f8cc7-5077-46bd-a6d4-c3cd6d67c9ff.png)
missing subtitle of the page "Configure services"

![image](https://user-images.githubusercontent.com/32871167/144392405-52edb15d-fe5f-4fb7-bc0d-be4c8e10c886.png)
On this page wrong sub title and missing info box
![image](https://user-images.githubusercontent.com/32871167/144392487-d912f544-40c0-403c-b284-e70c65d7aec8.png)

xkostka2 reviewed (changes requested):
![image](https://user-images.githubusercontent.com/47856862/145025019-0de987c8-7105-4d61-93d9-3038f90af817.png)
I managed to open disabled accordion. I entered Summary, with a destination selected, opened accordion, removed destination and the accordion in Summary remained open. 
I suggest closing accordions when leaving Summary. Also, please when accordion is disabled, change the cursor on-hover from pointer to classic cursor.

![image](https://user-images.githubusercontent.com/47856862/145026093-637a5ee2-3dcf-4477-a463-a11d7ee1fd89.png)
Question: Why isn't there an option to keep the facility as it is in previous tabs? I know there was some reason for it but I can't remember what was it.

HejdaJakub reviewed (changes requested):
- When some service is selected, but `NO` service destination is added, then the service destinations are disabled on summary. Other empty parts of summary are enabled to show, so it is little inconsistent.
- Please block `Create & configure` button for `As copy of` other facility. We will probably implement this functionality further, but now this combination throws an error, so we should disable this option in this moment.
- Now is very easy to mistakenly close the configuratoin facility guide without any feedback by clicking to left menu. I am not sure how difficult will be to manage the side menu, but if it is possible, there would be cool to open dialog (the same as is shown after click on cancel button) after click on some item of side menu, when user is still in configuration process.

bodnara commented:
Fixed accordion behaviour (consistency, cursor). 
Disabled configuration when copying facility. 
Added deactivate guard to prevent accidental configuration exit.

HejdaJakub commented:
> Question: Why isn't there an option to keep the facility as it is in previous tabs? I know there was some reason for it but I can't remember what was it.
>
It's because you have a `Finish` button on the `Summary` page and if you show button `Keep facility` in this dialog, these buttons (`Finish` and `Keep facility`) would do the same thing, but the different lables can be misleading. If you change lable in the dialog to `Finish`, it can be weird that you have `Finish` button on the `Summary` page and also in the cancel dialog and user could expect different behavior after clicking on these two buttons. But these buttons have the same behavior, so the button in the cancel dialog is duplicate. @xkostka2

HejdaJakub reviewed (changes requested):
I found one more broken use case. If I click at step 3 Services on `Next` and `Skip` (no service selected), then click on `Next` from step 6 Managers to step 7 Summary and then click on `Back` to step 6 Managers, it will break the `Next` button (nothing happens when I try to click on it) and also accordion is weird (step 3 Services looks like disabled). Otherwise I think it looks really good.
![image](https://user-images.githubusercontent.com/70371638/145834648-b3235900-1e7e-4a95-9c7a-62f4011bcba7.png)

xkostka2 reviewed (changes requested):
![image](https://user-images.githubusercontent.com/47856862/145837438-3f7d5460-34b7-40b3-97f4-0dbf1fe2e387.png)

Sorry that I forgot to mention it in the previous review. When there are no hosts or owners, the accordion isn't disabled and when there are no services then the accordion is disabled. Can you unify this, please? Disabling accordions of configurations when there are no services is ok for me.

xkostka2 reviewed (changes requested):
Sorry for another thing I missed before. I think we shouldn't allow users to remove themselves from role facility admin when configuring a facility or at least show them some warning that they won't be able to manipulate with facility and terminate the facility configuration.

bodnara commented:
> Sorry for another thing I missed before. I think we shouldn't allow users to remove themselves from role facility admin when configuring a facility or at least show them some warning that they won't be able to manipulate with facility and terminate the facility configuration.

We moved this step as last to prevend this happening accidentally. However I believe we wanted to allow this if it's intendencional. Mostly for the use case when our support will be configuring facility for someone and would want to remove themself at the end.
